app/scripts/metamask-controller.js

```diff
@@ -4995,6 +4996,27 @@ export default class MetamaskController extends EventEmitter {
     return null;
   }
 
+  /**
+   * Whether or not should skips publishing the transaction.
+   *
+   * @param {string} txMeta - The transaction meta.
+   * @param signedEthTx - Signed ethereum transaction.
+   */
+  // eslint-disable-next-line no-unused-vars
+  _shouldDisablePublish(txMeta, signedEthTx) {
```

Comment:
Rather than putting this logic and other MMI callbacks in here and bloating `MetaMaskController`, is it cleaner to put them in a separate file like `app/scripts/lib/mmi-hooks.ts`?

Reply:
Moved hooks to `app/scripts/lib/mmi-hooks.ts` and added unit test.